File: Eunleelee/2024-02/2월 15일/큰 수 만들기.py
def solution(number, k):
    answer = []

    for i in number:
        if not answer:
            answer.append(i)
            continue
        while answer[-1] < i and k > 0:
            answer.pop()
            k -= 1
            if not answer or k <= 0:
                break
        answer.append(i)
        if len(answer) == len(number) - k:
            break
    return ''.join(answer)

File: yht0827/2월/2월 15일/큰 수 만들기.java
class Solution {
	public String solution(String number, int k) {
		String answer = "";
		int ansLen = number.length() - k;
		int idx = 0;
		int remem = 0;
		for (int i = answer.length(); i < ansLen; i++) {
			char max = '0';
			for (int j = idx; j < number.length() - ansLen + answer.length() + 1; j++) {
				if (max < number.charAt(j)) {
					max = number.charAt(j);
					remem = j + 1;
				}
			}
			idx = remem;
			answer += max;
		}
		return answer;
	}
}